modules/_labs/pagination/react/lib/Pagination/common/Box.tsx

```diff
@@ -0,0 +1,46 @@
+/** @jsx jsx */
+import * as React from 'react';
+import {jsx, css} from '@emotion/core';
+import styled from '@emotion/styled';
+
+import {space, SpaceProps} from './utils/space';
+import {layout, LayoutProps} from './utils/layout';
+import {safelySpreadProps} from './utils/safelySpreadProps';
+
+const getBoxStyles = (props: BoxProps) => {
+  let styleProps = {};
+  for (const key in props) {
+    if (key in props) {
+      if (key in space) {
+        const value = props[key as keyof BoxProps];
+        const styleFn = space[key as keyof SpaceProps];
+        const styles = styleFn(value);
+        styleProps = {...styleProps, ...styles};
+      }
+      if (key in layout) {
+        const attr = layout[key as keyof LayoutProps];
+        styleProps = {...styleProps, [attr]: props[key as keyof BoxProps]};
+      }
+    }
+  }
+  return styleProps;
+};
+
+type As = keyof JSX.IntrinsicElements;
+type BoxRef = React.Ref<any> | null; // 'any' is appropriate here as Box can be composed into any element
+
+export interface BoxProps extends SpaceProps, LayoutProps, React.HTMLAttributes<HTMLElement> {
+  as?: As;
+  ref?: BoxRef;
+}
+
+export const StyledBox = styled('div')<BoxProps>({
+  boxSizing: 'border-box',
+});
+
+export const Box = ({as = 'div', ...props}: BoxProps) => {
+  // TODO: Memoize style props with React.useMemo
+  const boxStyles = getBoxStyles(props);
+  const safeProps = safelySpreadProps(props);
+  return <StyledBox as={as} css={css(boxStyles)} {...safeProps} />;
+};
```

Comment:
This isn't how refs work in React. Drop the `BoxRef` and `ref` here entirely and do this instead:

```suggestion
type As = keyof JSX.IntrinsicElements;

export interface BoxProps extends SpaceProps, LayoutProps, React.HTMLAttributes<HTMLElement> {
  as?: As;
}

export const StyledBox = styled('div')<BoxProps>({
  boxSizing: 'border-box',
});

export const Box = React.forwardRef<any, BoxProps>(({as = 'div', ...props}, ref) => {
  // TODO: Memoize style props with React.useMemo
  const boxStyles = getBoxStyles(props);
  const safeProps = safelySpreadProps(props);
  return <StyledBox ref={ref} as={as} css={css(boxStyles)} {...safeProps} />;
});
```

This will forward the provided `ref` properly (otherwise it will never ref to the element or anything for that matter).

This is the simplest way, but not the most correct way. To be more correct, you'll have to use something like the `Component` interface:

https://github.com/Workday/canvas-kit/blob/87a8a0179be158c5401cddc790a1d8e2b00f1e98/modules/common/react/lib/utils/components.ts#L19-L36